pkg/search/postgres/query/array_query.go

```diff
@@ -22,15 +21,15 @@ func getStringArrayPostTransformFunc(entry *QueryEntry) (func(val interface{}) i
 		return nil, errors.New("no filter func found")
 	}
 	return func(val interface{}) interface{} {
-		textArray, _ := val.(*pgtype.TextArray)
+		textArray, _ := val.(*[]string)
 		if textArray == nil {
-			return (*pgtype.TextArray)(nil)
+			return nil
```

Comment:
Why did you change the return value from typed nil to untyped nil? This changes the behavior if the return value is nil checked.

Reply:
Good point, fixed